frontend/src/metabase/query_builder/components/notebook/steps/LimitStep/LimitStep.tsx

```diff
@@ -16,15 +17,19 @@ function LimitStep({
   const { stageIndex } = step;
 
   const limit = Lib.currentLimit(query, stageIndex);
-  const value = typeof limit === "number" ? limit : "";
+  const [value, setValue] = useState(typeof limit === "number" ? limit : "");
 
-  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const nextLimit = parseInt(e.target.value, 0);
+  const handleBlur = (event: FocusEvent<HTMLInputElement>) => {
+    const nextLimit = parseInt(event.target.value, 0);
```

Comment:
hmm, radix must be `10` here except we want to support non-decimal numbers
```suggestion
    const nextLimit = parseInt(event.target.value, 10);
```